src/sched.rs

```diff
@@ -215,6 +215,112 @@ mod sched_linux_like {
 
         Errno::result(res).map(drop)
     }
+
+    libc_bitflags! {
+        pub struct SchedFlags: c_int {
+            #[cfg(target_os = "android")]
+            SCHED_NORMAL;
+            #[cfg(target_os = "linux")]
+            SCHED_OTHER;
+            SCHED_FIFO;
+            SCHED_RR;
+            SCHED_BATCH;
+            SCHED_IDLE;
+            #[cfg(target_os = "android")]
+            SCHED_DEADLINE;
+            #[cfg(target_os = "linux")]
+            SCHED_RESET_ON_FORK;
+        }
+    }
+
+    #[repr(C)]
+    #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+    pub struct SchedParam(libc::sched_param);
+
+    impl SchedParam {
+        pub fn new(priority: i32) -> Self {
+            let mut sched_param: libc::sched_param = unsafe { mem::zeroed() };
+            sched_param.sched_priority = priority;
+            SchedParam(sched_param)
+        }
+
+        pub fn priority(&self) -> i32 {
+            self.0.sched_priority
+        }
+    }
+
+    impl Default for SchedParam {
+        fn default() -> Self {
+            SchedParam::new(0)
+        }
+    }
+
+    /// Get minimum priority value for policy
+    ///
+    /// See also [`sched_get_priority_min(2)`](https://man7.org/linux/man-pages/man2/sched_get_priority_min.2.html)
+    pub fn sched_get_priority_min(policy: SchedFlags) -> Result<i32> {
+        let res = unsafe { libc::sched_get_priority_min(policy.bits()) };
+
+        Errno::result(res)
+    }
+
+    /// Get maximum priority value for policy
+    ///
+    /// See also [`sched_get_priority_max(2)`](https://man7.org/linux/man-pages/man2/sched_get_priority_max.2.html)
+    pub fn sched_get_priority_max(policy: SchedFlags) -> Result<i32> {
+        let res = unsafe { libc::sched_get_priority_max(policy.bits()) };
+
+        Errno::result(res)
+    }
+
+    /// Set thread's scheduling parameters
+    ///
+    /// `pid` is the thread ID to update.
+    /// If `pid` is zero, then the parameters for the calling thread are set.
+    ///
+    /// See also [`sched_setparam(2)`](https://man7.org/linux/man-pages/man2/sched_setparam.2.html)
+    pub fn sched_setparam(pid: Pid, sched_param: &SchedParam) -> Result<()> {
```

Comment:
If pid 0 has a special meaning for the C function, then a Rustier way to bind is is for the argument to be of type `Option<Pid>`.  See for example https://docs.rs/nix/0.20.0/nix/unistd/fn.getsid.html .

Reply:
I agree that's a Rustier way to do it. That said, other functions in `nix::sched` such as [`sched_setaffinity`](https://docs.rs/nix/0.20.0/nix/sched/fn.sched_setaffinity.html) take raw `Pid` types -- I based most of the implementation for the new `sched_*` functions off of the existing ones.

I've got no problem switching to `Option<Pid>` here, but thought I'd point out the inconsistency before making changes.